### PR TITLE
Remote repository: save and set default_branch

### PR DESCRIPTION
**Pull request by stsewd**
Currently we default to master/trunk when a project is imported,
even when the project isn't imported manually.

The default_branch is already in the API response of all our supported
providers.

Questions:

- Should we show this in the first form or in the advanced form?

Close https://github.com/readthedocs/readthedocs.org/issues/7533
Close https://github.com/readthedocs/readthedocs.org/issues/7601

### CODE REVIEW

**Comment by saadmk11** [comment]
https://github.com/readthedocs/readthedocs.org/pull/7536/ needs to be updated after this PR is merged as a new migration is added here :)

File: readthedocs/projects/forms.py [hunk 1]
```diff
@@ -166,6 +166,7 @@ class Meta:
         fields = (
             'description',
             'documentation_type',
+            'default_branch',
```
Comment (ericholscher) [inline readthedocs/projects/forms.py:169]: Is this to allow users to edit this in the Project edit form, or is it required for the `import.js` to pass it properly?

**Review by ericholscher — APPROVED** [review]
This seems like a good solution to users having other kinds of default branches. Just one question, but it seems like we should get this shipped to fix users with `main`.

File: readthedocs/projects/forms.py (hunk 1 above)
Comment (ericholscher) [reply readthedocs/projects/forms.py:169]: Ah, this is on import. I'm 👍  on showing it only when users select "Edit advanced options" 👍

Comment (stsewd) [reply readthedocs/projects/forms.py:169]: For both, yeah p:

**Pull request merged**